### DISCUSSION TITLE
v6 changing how version values are displayed?

### DISCUSSION THREAD

**Discussion by mharding-hpe** [GitTools/GitVersion#4128] [Q&A] [answered]
v5:
```json
{
  "Major": 2,
  "Minor": 22,
  "Patch": 0,
  "PreReleaseTag": "casmcms-8694.1",
  "PreReleaseTagWithDash": "-casmcms-8694.1",
  "PreReleaseLabel": "casmcms-8694",
  "PreReleaseLabelWithDash": "-casmcms-8694",
  "PreReleaseNumber": 1,
  "WeightedPreReleaseNumber": 30001,
  "BuildMetaData": 34,
  "BuildMetaDataPadded": "0034",
  "FullBuildMetaData": "34.Branch.feature-casmcms-8694.Sha.c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "MajorMinorPatch": "2.22.0",
  "SemVer": "2.22.0-casmcms-8694.1",
  "LegacySemVer": "2.22.0-casmcms-8694-1",
  "LegacySemVerPadded": "2.22.0-casmcms-8694-0001",
  "AssemblySemVer": "2.22.0.0",
  "AssemblySemFileVer": "2.22.0.0",
  "FullSemVer": "2.22.0-casmcms-8694.1+34",
  "InformationalVersion": "2.22.0-casmcms-8694.1+34.Branch.feature-casmcms-8694.Sha.c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "BranchName": "feature/casmcms-8694",
  "EscapedBranchName": "feature-casmcms-8694",
  "Sha": "c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "ShortSha": "c0c728c",
  "NuGetVersionV2": "2.22.0-casmcms-8694-0001",
  "NuGetVersion": "2.22.0-casmcms-8694-0001",
  "NuGetPreReleaseTagV2": "casmcms-8694-0001",
  "NuGetPreReleaseTag": "casmcms-8694-0001",
  "VersionSourceSha": "a4c125f010957d947e153fca93532dae2092d568",
  "CommitsSinceVersionSource": 34,
  "CommitsSinceVersionSourcePadded": "0034",
  "UncommittedChanges": 0,
  "CommitDate": "2024-07-15"
}
```

v6:

```json
{
  "AssemblySemFileVer": "2.23.0.0",
  "AssemblySemVer": "2.23.0.0",
  "BranchName": "feature/casmcms-8694",
  "BuildMetaData": null,
  "CommitDate": "2024-07-15",
  "CommitsSinceVersionSource": 8,
  "EscapedBranchName": "feature-casmcms-8694",
  "FullBuildMetaData": "Branch.feature-casmcms-8694.Sha.c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "FullSemVer": "2.23.0-{BranchName}.8",
  "InformationalVersion": "2.23.0-{BranchName}.8+Branch.feature-casmcms-8694.Sha.c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "Major": 2,
  "MajorMinorPatch": "2.23.0",
  "Minor": 23,
  "Patch": 0,
  "PreReleaseLabel": "{BranchName}",
  "PreReleaseLabelWithDash": "-{BranchName}",
  "PreReleaseNumber": 8,
  "PreReleaseTag": "{BranchName}.8",
  "PreReleaseTagWithDash": "-{BranchName}.8",
  "SemVer": "2.23.0-{BranchName}.8",
  "Sha": "c0c728cf8579d03a11fb56db0348bbfcb04eb85a",
  "ShortSha": "c0c728c",
  "UncommittedChanges": 1,
  "VersionSourceSha": "7b90ab2149b66f959aa3dbf28dfc4d3daa38af58",
  "WeightedPreReleaseNumber": 30008
}
```

In v5, the branch name value was automatically filled in where appropriate. In v6, it always shows `{BranchName}`. This has broken some of our CI/CD tooling, which calls gitversion to retrieve the `PreReleaseTag`. 

Is this how v6 is supposed to work? I don't see any CLI option I can specify to tell it to substitute the actual values like it did previously. I looked at the v6 breaking change notes and didn't see this mentioned, but I wasn't sure if it was implied by one of the things listed, and I failed to understand it.

**Comment by mharding-hpe** [comment]
I had been using `/showvariable` to get the individual values, and I thought maybe switching to `/format` would do the trick, but it gives the same output.

**Comment by asbjornu** [comment]
Hmyea, this seems like a bug to me. @HHobeck thoughts?

**Reply by HHobeck** [reply, marked as answer]
Hi there. 

In the breaking change documentation the following is written:

> * The `BranchPrefixToTrim` configuration property has been removed. `RegularExpression` is now used to capture named groups instead.
>   * Default `RegularExpression` for feature branches is changed from `^features?[/-]` to `^features?[/-](?<BranchName>.+)` to support using `{BranchName}` out-of-the-box
>   * Default `RegularExpression` for unknown branches is changed from `.*` to `(?<BranchName>.+)` to support using `{BranchName}` out-of-the-box

I think you have not specified the capturing group with name `BranchName` in the `regex` property. However have you tried this without using a customized configuration?

```yaml
workflow: GitFlow/v1
```

Happy Branching!

**Reply by mharding-hpe** [reply]
I think you have identified the problem! This is the regex used for the feature branches:

```yaml
    regex: ^features?[/-]
```

As you suspected, no capturing group defined. I'll make the change you suggest and confirm that it fixes the issue.

Thanks for the quick response!

**Comment by mharding-hpe** [comment]
I have confirmed that changing the regex fixed the issue for me. Thanks again!
